lib/cfg-block.c

```diff
@@ -61,6 +61,50 @@ cfg_block_format_name(CfgBlockGenerator *s, gchar *buf, gsize buf_len)
   return buf;
 }
 
+/* token block args */
+
+static void
+_validate_args_callback(gpointer k, gpointer v, gpointer user_data)
+{
+  CfgArgs *defs = ((gpointer *) user_data)[0];
+  const gchar *reference = ((gpointer *) user_data)[1];
+  gpointer *problem_found = &((gpointer *) user_data)[2];
+
```

Comment:
As I see problem_found is write-only now? What is the purpose? Is an early return missing in this function?

Reply:
Ok I see it is used on line 100 with `if (validate_params[2])`. The early early return might still be useful.

Reply:
what do you mean on early return? the entire function is just one if statement, the iteration happens in the foreach which uses this as a callback. You think that the return would make this more readable?

Reply:
I was thinking if the previous iteration ran into an error, then we could skip the execution of this function if we return when problem_found is true.

Maybe it would be better use g_hash_table_find instead of g_hash_table_foreach to avoid unnecessary calls. With g_hash_table_find, probably we could eliminate the problem_found variable.

Though I agree this is not a performance heavy path, so we can stick with anything that is more readable than optimal.